src/meshapi/views/helpers.py

```diff
@@ -0,0 +1,128 @@
+                                    additional_keys=("network_number",),
+                                    help_text="A Node with the network number exactly matching the requested number, "
+                                    "if it exists",
+                                ),
+                                "exact_match_nonrecycled_install": InstallNestedRefSerializer(
+                                    help_text="An install  with the install number exactly matching the requested "
+                                    "number, if that install has NOT had its install number recycled (or null if "
+                                    "none exists)"
+                                ),
+                            },
+                        ),
+                    },
+                ),
+                description="At least one Node or Install exists corresponding to this",
+            ),
+            "400": OpenApiResponse(
+                helper_err_response_schema,
+                description="Invalid request",
+            ),
+            "404": OpenApiResponse(
+                helper_err_response_schema,
+                description="Requested number could not be found as either an Network or Install number",
+            ),
+            "500": OpenApiResponse(helper_err_response_schema, description="Unexpected internal error"),
+        },
+    )
+    def get(self, request: Request) -> Response:
+        ambiguous_number_str = request.query_params.get("number", "")
+        try:
+            ambiguous_number = int(ambiguous_number_str)
+            if ambiguous_number <= 0:
+                raise ValueError()
+        except ValueError:
+            return Response(
+                {"detail": f"Invalid number: '{ambiguous_number_str}'. Must be an integer greater than zero"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        install_object = Install.objects.filter(install_number=ambiguous_number).first()
+        node_object = Node.objects.filter(network_number=ambiguous_number).first()
+
+        if not install_object and not node_object:
+            return Response(
+                {"detail": f"Provided number: {ambiguous_number} did not correspond to any install or node objects"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        resolved_node = node_object if node_object else (install_object.node if install_object else None)
+
+        output = {
+            "resolved_node": node_serializer_field.to_representation(resolved_node) if resolved_node else None,
+            "supporting_data": {
+                "exact_match_recycled_install": (
+                    install_serializer_field.to_representation(install_object)
+                    if install_object and install_object.status == Install.InstallStatus.NN_REASSIGNED
+                    else None
+                ),
+                "exact_match_nonrecycled_install": (
```

Comment:
Nit, I think it would be clearer to have `exact_match_install`:
```suggestion
                "exact_match_install": (
```